### PR TITLE
Refactoring banking system

### PR DESCRIPTION
**Pull request by DaniilYusha**

### CODE REVIEW

File: entities/usual_card.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  def withdraw_tax(amount)
+    tax(amount, withdraw_percent, withdraw_fixed)
+  end
+
+  def put_tax(amount)
+    tax(amount, put_percent, put_fixed)
+  end
+
+  def sender_tax(amount)
+    tax(amount, sender_percent, sender_fixed)
```
Comment (Anastasiia04) [inline entities/usual_card.rb]: the same def in base class?

File: entities/virtual_card.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    tax(amount, withdraw_percent, withdraw_fixed)
+  end
+
+  def put_tax(amount)
+    tax(amount, put_percent, put_fixed)
+  end
+
+  def sender_tax(amount)
+    tax(amount, sender_percent, sender_fixed)
+  end
```
Comment (Anastasiia04) [inline entities/virtual_card.rb]: the same

File: modules/account_manager.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      FileLoader.save new_accounts
+    end
+
+    def find_card(number)
+      all_cards.find { |card| card.number == number }
+    end
+
+    def card_exists?(number)
+      all_cards.select { |card| card.number == number }.any?
+    end
```
Comment (Anastasiia04) [inline modules/account_manager.rb]: seems the same
maybe leave only the def find_card and use find_card.nil?  if we need to check for a card exist?

File: modules/validator.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module Validator
```
Comment (Anastasiia04) [inline modules/validator.rb]: we use this only in Account class?
maybe move this one def to class?

**Review by Anastasiia04 — COMMENTED** [review]
Seems nice!

File: entities/account.rb [hunk 5]
```diff
@@ -0,0 +1,65 @@
+class Account
+  include Validator
+
+  attr_accessor :errors
```
Comment (vasivaas) [inline entities/account.rb]: why attr_accessor? maybe you can change to attr_reader?

**Review by vasivaas — APPROVED** [review]
nice!

File: entities/usual_card.rb (hunk 1 above)
Comment (DaniilYusha) [reply entities/usual_card.rb]: @Anastasiia04 no, in BaseCard class I have methods like `put_money`, `send_money`, but here only tax calculation

File: modules/account_manager.rb (hunk 3 above)
Comment (DaniilYusha) [reply modules/account_manager.rb]: @Anastasiia04 ok, done

File: entities/account.rb (hunk 5 above)
Comment (DaniilYusha) [reply entities/account.rb]: @vasivaas thanks, done

File: modules/account_registrator.rb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    outputer.enter_name
+    gets.chomp
+  end
+
+  def age_input
+    outputer.enter_age
+    gets.chomp.to_i
+  end
+
+  def login_input
```
Comment (vasivaas) [inline modules/account_registrator.rb]: name_input, login_input, password input are identical, differ only in the initial message, it is possible to make a unique method which will have 1 parameter - the initial message?

File: modules/account_actions.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+module AccountActions
```
Comment (vasivaas) [inline modules/account_actions.rb]: maybe you can move this method to another module? i think creating a module for one method  is not a best practice

**Review by vasivaas — CHANGES_REQUESTED** [review]
🔥 !

**Review by Anastasiia04 — APPROVED** [review]
good job!

**Review by vasivaas — APPROVED** [review]
🚀

**Review by kolyanjy — APPROVED** [review]

**Pull request merged**